homeassistant/components/homekit_controller/diagnostics.py

```diff
@@ -66,7 +66,7 @@ def _async_get_diagnostics_for_device(
         state = hass.states.get(entity_entry.entity_id)
         state_dict = None
         if state:
-            state_dict = async_redact_data(state.as_dict(), REDACTED_STATE)
+            state_dict = cast(dict, async_redact_data(state.as_dict(), REDACTED_STATE))
```

Comment:
I think the typing changes in https://github.com/home-assistant/core/commit/cc6b0cc84394f78a326624eb503bf0517f1bdbf4 are wrong. `async_redact_data` *is* returning a `dict`, not a `MappingProxyType`, so the cast here is papering over a bug in `async_redact_data` IMO.

Reply:
async_redact_data will convert any type inheriting from `Mapping` to a dict. It's typed to have input = output type.

Let me see if I can fix that.

Reply:
Initial attempt gives me an error.

```python
@overload
def async_redact_data(data: Mapping, to_redact: Iterable[Any]) -> dict:
    ...


@overload
def async_redact_data(data: T, to_redact: Iterable[Any]) -> T:
    ...
```

homeassistant/components/diagnostics/util.py:15: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
Found 1 error in 1 file (checked 2 source files)

Reply:
ignored that error 🤷